src/Blocks/components/form/assets-admin/increment.js

```diff
@@ -0,0 +1,70 @@
+export class Increment {
+	constructor(options = {}) {
+		/** @type {import('../assets/utils').Utils} */
+		this.utils = options.utils;
+		/** @type {import('../assets/state').State} */
+		this.state = this.utils.getState();
+
+		this.selector = options.selector;
+		this.confirmMsg = options.confirmMsg;
+	}
+
+	init() {
+		[...document.querySelectorAll(this.selector)].forEach((element) => {
+			element.addEventListener('click', this.onClick, true);
+		});
+	}
+
+	// Handle form submit and all logic.
+	onClick = (event) => {
+		event.preventDefault();
+
+		if(!confirm(this.confirmMsg)) {
+			return;
+		}
+
+		const formId = this.state.getFormIdByElement(event.target);
+		const field = this.state.getFormFieldElementByChild(event.target);
+
+		const formData = new FormData();
+
+		formData.append('formId', field.getAttribute(this.state.getStateAttribute('formId')));
+		this.utils.showLoader(formId);
+
+		// Populate body data.
+		const body = {
+			method: 'POST',
+			mode: 'same-origin',
+			headers: {
+				Accept: 'application/json',
+				'X-WP-Nonce': this.state.getStateConfigNonce(),
+			},
+			body: formData,
+			credentials: 'same-origin',
+			redirect: 'follow',
+			referrer: 'no-referrer',
+		};
```

Comment:
```
// Populate body data.
		const body = {
			method: 'POST',
			headers: {
				Accept: 'application/json',
				'X-WP-Nonce': this.state.getStateConfigNonce(),
			},
			body: formData,
			redirect: 'follow',
			referrer: 'no-referrer',
		};
```

`same-origin` is default here so I don't think you need this stated.

Reply:
tnx